4.analyze-components/scripts/nbconverted/3.within-z-svcca-analysis.py

```diff
@@ -0,0 +1,93 @@
+
+# coding: utf-8
+
+# # Determining Within Z Dimension Stability with SVCCA
+# 
+# **Gregory Way 2018**
+# 
+# Here, we apply Singular Vector Canonical Correlation Analysis ([Raghu et al. 2017](https://arxiv.org/abs/1706.05806 "SVCCA: Singular Vector Canonical Correlation Analysis for Deep Learning Dynamics and Interpretability")) ([github](https://github.com/google/svcca)) to the neuron matrices Z to quantify model stability both within and across algorithms over each bottleneck dimensionality.
+# 
+# Briefly, SVCCA uses Singular Value Decomposition (SVC) to extract the components explaining 99% of the variation.
```

Comment:
why 99%?

Reply:
in the original [paper](https://arxiv.org/abs/1706.05806) filtering out the 1% seemed to rid most nuisance factors from causing spurious correlations:

> In general we take enough directions to explain 99% of variance in the
subspace. This is especially important in neural network representations, where as we will
show many low variance directions (neurons) are primarily noise.